lib/CAPI/TTNNAttrs.cpp

```diff
@@ -69,14 +68,26 @@ MlirAttribute ttmlirTTNNMeshShapeAttrGet(MlirContext ctx, int64_t y,
   return wrap(MeshShapeAttr::get(unwrap(ctx), y, x));
 }
 
-MlirAttribute ttmlirTTNNTTNNLayoutAttrGet(MlirContext ctx, MlirAffineMap linear,
-                                          MlirAttribute grid, MlirType memref,
-                                          unsigned memLayout) {
+// Get layout TTNNLayout attribute
+//
+// param ctx: mlir context
+// param linear Affine map for mapping tensor from logical to physical space
+// param grid Grid of cores where tensor is mapped to
+// param memref Memref which holds shard size, shard scalar type and memory
+// space param memLayout Optional memory layout
+MlirAttribute
+ttmlirTTNNTTNNLayoutAttrGet(MlirContext ctx, MlirAffineMap linear,
+                            MlirAttribute grid, MlirType memref,
+                            std::optional<unsigned> memLayout = std::nullopt) {
```

Comment:
yea it shouldnt look like this, thanks for pointing out... I can default it to some value for cases where memLayout is not needed. I need some more time to think how this API should look like. @nsmithtt since you added this, is it fine to default to uint_max for cases where we don't want tensormemlayout? I will create separate item to change this API and to add unit test which will test all APIs in CAPI lib?